slick.core.js

```diff
@@ -675,10 +721,239 @@
   function regexSanitizer(dirtyHtml) {
      return dirtyHtml.replace(/(\b)(on[a-z]+)(\s*)=|javascript:([^>]*)[^>]*|(<\s*)(\/*)script([<>]*).*(<\s*)(\/*)script(>*)|(&lt;)(\/*)(script|script defer)(.*)(&gt;|&gt;">)/gi, '');
   }
- 
-  // exports
-  $.extend(true, window, {
-    "Slick": {
+
+  // With help from https://youmightnotneedjquery.com/
+  function grep( elems, callback, invert ) {
+		var callbackInverse,
+			matches = [],
+			i = 0,
+			length = elems.length,
+			callbackExpect = !invert;
+
+		// Go through the array, only saving the items
+		// that pass the validator function
+		for ( ; i < length; i++ ) {
+			callbackInverse = !callback( elems[ i ], i );
+			if ( callbackInverse !== callbackExpect ) {
+				matches.push( elems[ i ] );
+			}
+		}
+
+		return matches;
+	}
+
+  function offset(el) {
+
+    box = el.getBoundingClientRect();
+    docElem = document.documentElement;
+    return {
+      top: box.top + window.pageYOffset - docElem.clientTop,
+      left: box.left + window.pageXOffset - docElem.clientLeft
+    };
+ 	}
+
+  function width(el, value) {
+    if (value === undefined) {
+      return el.getBoundingClientRect().width;
+    }
+    setStyleSize(el, "width", value);
+  }
+
+  function height(el, value) {
+    if (value === undefined) {
+      return el.getBoundingClientRect().height;
+    }
+    setStyleSize(el, "height", value);
+  }
+
+  function setStyleSize(el, style, val) {
+    if (typeof val === 'function') val = val();
+    if (typeof val === 'string') el.style[style] = val;
+    else el.style[style] = val + 'px';
```

Comment:
I'm a big fan of wrapping everything in curly braces to avoid potential future bugs, it would also make the code clearer and easier to read

Reply:
Done